controllers/config_helper.go

```diff
@@ -0,0 +1,249 @@
+	cacheRuntime "sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+// Default config data
+func getDefaultConfig(namespace string) *v1alpha1.Config {
+	config := &v1alpha1.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "config",
+		},
+		Spec: v1alpha1.ConfigSpec{
+			Match: []v1alpha1.MatchEntry{
+				{
+					ExcludedNamespaces: []wildcard.Wildcard{
+						"kube-*", "multicluster-engine",
+						"hypershift", "hive", "rhacs-operator", "open-cluster-*",
+					},
+					Processes: []string{
+						"audit", "webhook", "sync",
+					},
+				},
+			},
+		},
+	}
+
+	return config
+}
+
+func createDefaultConfig(ctx context.Context, c client.Client, namespace string,
+	gatekeeper *operatorv1alpha1.Gatekeeper, scheme *runtime.Scheme,
+) error {
+	var config *v1alpha1.Config
+
+	// When it is DisableDefaultMatches = true
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.DisableDefaultMatches != nil &&
+		*gatekeeper.Spec.Config.DisableDefaultMatches {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: namespace,
+			},
+		}
+	} else {
+		config = getDefaultConfig(namespace)
+	}
+
+	matches := getMatches(gatekeeper)
+
+	// Append matched from gatekeeper cr spec.config.matches
+	config.Spec.Match = append(config.Spec.Match, matches...)
+
+	// Add ownerReference
+	if err := controllerutil.SetControllerReference(gatekeeper, config, scheme); err != nil {
+		return err
+	}
+
+	err := c.Create(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// it returns gatekeeper.spec.config.matches
+func getMatches(gatekeeper *operatorv1alpha1.Gatekeeper) (matches []v1alpha1.MatchEntry) {
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.Matches != nil {
+		// Will append gatekeeper.Spec.Config.Matches
+		matches = gatekeeper.Spec.Config.Matches
+	}
+
+	return matches
+}
+
+func hasCurrentGatekeeperUid(existingConfig *v1alpha1.Config, gatekeeper *operatorv1alpha1.Gatekeeper) bool {
+	for _, or := range existingConfig.OwnerReferences {
+		if or.UID == gatekeeper.UID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Delete previous gatekeeper ownerReference
+func deletePrevOwnerReference(existingConfig v1alpha1.Config) []metav1.OwnerReference {
+	index := -1
+
+	for i, or := range existingConfig.OwnerReferences {
+		if or.Name == "gatekeeper" {
+			index = i
+
+			break
+		}
+	}
+
+	if index == -1 {
+		return existingConfig.OwnerReferences
+	}
+
+	return append(existingConfig.OwnerReferences[:index], existingConfig.OwnerReferences[index+1:]...)
+}
+
+// Reset Config.Spec.Match and append the default exempt namespaces and
+// provided matched from gatekeeper CR
+func addDefaultExemptNamespaces(ctx context.Context,
```

Comment:
The name doesn't really fit after you refactored PR. How about this?
```suggestion
func setExemptNamespaces(ctx context.Context,
```